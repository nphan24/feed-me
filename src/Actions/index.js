export const postRecipes = recipes => ({
  type: 'POST_RECIPES',
  recipes
});

export const replaceRecipes = recipes => ({
  type: 'REPLACE_RECIPES',
  recipes
});

export const addFavorite = recipe => ({
  type: 'ADD_FAVORITE',
  recipe
});

export const removeFavorite = id => ({
  type: 'REMOVE_FAVORITE',
  id
});

export const addUser = user => ({
  type: 'ADD_USER',
  user
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});

export const promptSignIn = error => ({
  type: 'PROMPT_SIGNIN',
  error
});