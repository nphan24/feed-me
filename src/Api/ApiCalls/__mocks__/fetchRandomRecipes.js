export const fetchRandomRecipes = jest.fn().mockImplementation(() => [
  {
    Ingredients:[
      'butter',
      'heavy cream',
      'salt',
      'ground black pepper',
      'russet potatoes',
      'parmigiano'
    ],
    id: 'Potatoes-Au-Gratin-2244864',
    image:
      'https://lh3.googleusercontent.com/1UaqgvAZrw346jUpat4-at6mkz6-Jhjymn8AkVY2Kwu48E0TXbuH9OnhsR_k5OF-5UV_MSxx6qbnFPnawzTtqQ=s90-c',
    name: 'Potatoes Au Gratin',
    source: 'https://www.onceuponachef.com/recipes/potatoes-au-gratin.html',
    totaltime: 75
  }
]);
