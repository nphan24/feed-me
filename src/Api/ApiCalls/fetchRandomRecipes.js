import { apiKey, applicationId } from './apiKey';
import { cleanRecipes } from 
  '../helper/cleanRecipes';
import { fetchRecipeSource } from './fetchRecipeSource';

export const fetchRandomRecipes = async (selected) => {
  try {
    let category = selected ? selected : '';
    // let category = 'chicken';
    console.log('select', category);
    const response = await fetch(`http://api.yummly.com/v1/api/recipes?_app_id=${applicationId}&_app_key=${apiKey}&q=${category}&maxResult=30&start=00`);
    const data = await response.json();
    console.log(data)
    const recipesWithoutSource = await cleanRecipes(data.matches);
    const recipesWithSource = await fetchRecipeSource(recipesWithoutSource);
    return recipesWithSource;
  } catch (error) {
    throw new error(error.message);
  }
};

