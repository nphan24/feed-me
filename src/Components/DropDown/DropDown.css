.dropdown-menu {
  color: black;
  z-index: 100;
  position: absolute;
  margin: 0;
}

ul li {
  display: none;
}