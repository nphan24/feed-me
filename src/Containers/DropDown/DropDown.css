.dropdown-menu {
  width: 250px;
  height: 35px;
  font-size: 24px;
}

@media only screen and (max-width: 400px) {
  .dropdown-menu {
    margin: auto;
  }
}
