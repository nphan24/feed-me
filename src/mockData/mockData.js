/* eslint-disable */

export const mockRecipe = {
  name: 'pasta',
  image: 'url',
  totalTime: '30',
  id: 2
};

export const uncleanedMockRecipeData = { matches: [
  { recipeName: 'tofu', 
    imageUrlsBySize: 'url', 
    ingredients: ['',''], 
    totalTimeInSeconds: 3600, 
    id: 2,
    pasta: 'speghetti',
    fruit: 'mango'
  }
]
 };

export const cleanedMockRecipeData = [{ 
  Ingredients: ['', ''], 
  id: 2, 
  image: undefined, 
  name: 'tofu', 
  totaltime: 60 
}];

export const mockRecipesArray = [
  {
    name: 'pasta',
    image: 'url',
    totalTime: '30',
    id: 2
  },
  {
    name: 'rice',
    image: 'hmm',
    totalTime: '90',
    id: 3
  }
];

export const mockRemainingArray = [
  {
    name: 'rice',
    image: 'hmm',
    totalTime: '90',
    id: 3
  }
];
