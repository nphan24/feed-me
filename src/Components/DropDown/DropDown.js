import React, {Component} from 'react';
import Dropdown from 'react-dropdown';
import './DropDown.css';
import { fetchRandomRecipes } from '../../Api/ApiCalls/fetchRandomRecipes';

export class DropDown extends Component {

  handleChange = (event) => {
    fetchRandomRecipes(event.target.value);
  };

  render() {
    return <div>
      <select id="dropdown" onChange={ this.handleChange}>
        <option value="Categories">Categories</option>
        <option value="breakfast">breakfast</option>
        <option value="lunch">lunch</option>
        <option value="dinner">dinner</option>
        <option value="appetizer">appetizer</option>
        <option value="chicken">chicken</option>
        <option value="beef">beef</option>
        <option value="lamb">lamb</option>
        <option value="seafood">seafood</option>
        <option value="pork">pork</option>
        <option value="smoothies">smoothies</option>
        <option value="snacks">snacks</option>
      </select>
    </div>;
  }
}