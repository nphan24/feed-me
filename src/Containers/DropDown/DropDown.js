import React, {Component} from 'react';
import './DropDown.css';
import { fetchRandomRecipes } from '../../Api/ApiCalls/fetchRandomRecipes';
import { connect } from 'react-redux';
import * as Actions from '../../Actions';
export class DropDown extends Component {

  handleChange = async (event) => {
    const recipes = await fetchRandomRecipes(event.target.value);
    this.props.replaceRecipes(recipes);
  };

  render() {
    return <div>
      <select id="dropdown" onChange=
        {this.handleChange}>
        <option value="Categories">Categories</option>
        <option value="breakfast">breakfast</option>
        <option value="lunch">lunch</option>
        <option value="dinner">dinner</option>
        <option value="appetizer">appetizer</option>
        <option value="chicken">chicken</option>
        <option value="beef">beef</option>
        <option value="lamb">lamb</option>
        <option value="seafood">seafood</option>
        <option value="pork">pork</option>
        <option value="smoothies">smoothies</option>
        <option value="snacks">snacks</option>
      </select>
    </div>;
  }
}

export const mapDispatchToProps = dispatch => ({
  replaceRecipes: recipes => dispatch(Actions.replaceRecipes(recipes))
});

export default (connect(null, mapDispatchToProps)(DropDown));