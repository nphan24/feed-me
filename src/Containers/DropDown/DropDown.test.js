import { DropDown, mapDispatchToProps} from './DropDown';
import * as mock from '../../mockData/mockData';
import { shallow } from 'enzyme';
import React from 'react';
import { fetchRandomRecipes } from '../../Api/ApiCalls/fetchRandomRecipes';
import * as Actions from '../../Actions';
jest.mock('../../Api/ApiCalls/fetchRandomRecipes');

describe('DropDown', () => {
  let wrapper;
  let mockreplaceRecipes = jest.fn();

  beforeEach(()=> {
    wrapper = shallow(<DropDown 
      replaceRecipes={mockreplaceRecipes}/>);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchRandomRecipes on handleChange', () => {
    // expect(fetchRandomRecipes).toHaveBeenCalled();
  });

  it('should call replaceRecipes on handleChange', () => {
    // wrapper.instance().handleChange();
    // expect(mockreplaceRecipes).toHaveBeenCalled();
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with the correct params on replaceRecipes', () => {
    const mockDispatch = jest.fn();
    const recipe = mock.mockRecipe;
    const expected = Actions.replaceRecipes(recipe);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.replaceRecipes(recipe);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});