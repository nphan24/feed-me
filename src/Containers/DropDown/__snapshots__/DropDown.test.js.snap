// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropDown
    replaceRecipes={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        id="dropdown"
        onChange={[Function]}
>
        <option
                value="Categories"
        >
                Categories
        </option>
        <option
                value="breakfast"
        >
                breakfast
        </option>
        <option
                value="lunch"
        >
                lunch
        </option>
        <option
                value="dinner"
        >
                dinner
        </option>
        <option
                value="appetizer"
        >
                appetizer
        </option>
        <option
                value="chicken"
        >
                chicken
        </option>
        <option
                value="beef"
        >
                beef
        </option>
        <option
                value="lamb"
        >
                lamb
        </option>
        <option
                value="seafood"
        >
                seafood
        </option>
        <option
                value="pork"
        >
                pork
        </option>
        <option
                value="smoothies"
        >
                smoothies
        </option>
        <option
                value="snacks"
        >
                snacks
        </option>
</select>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            value="Categories"
>
            Categories
</option>,
          <option
            value="breakfast"
>
            breakfast
</option>,
          <option
            value="lunch"
>
            lunch
</option>,
          <option
            value="dinner"
>
            dinner
</option>,
          <option
            value="appetizer"
>
            appetizer
</option>,
          <option
            value="chicken"
>
            chicken
</option>,
          <option
            value="beef"
>
            beef
</option>,
          <option
            value="lamb"
>
            lamb
</option>,
          <option
            value="seafood"
>
            seafood
</option>,
          <option
            value="pork"
>
            pork
</option>,
          <option
            value="smoothies"
>
            smoothies
</option>,
          <option
            value="snacks"
>
            snacks
</option>,
        ],
        "id": "dropdown",
        "onChange": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Categories",
            "value": "Categories",
          },
          "ref": null,
          "rendered": "Categories",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "breakfast",
            "value": "breakfast",
          },
          "ref": null,
          "rendered": "breakfast",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "lunch",
            "value": "lunch",
          },
          "ref": null,
          "rendered": "lunch",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "dinner",
            "value": "dinner",
          },
          "ref": null,
          "rendered": "dinner",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "appetizer",
            "value": "appetizer",
          },
          "ref": null,
          "rendered": "appetizer",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "chicken",
            "value": "chicken",
          },
          "ref": null,
          "rendered": "chicken",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "beef",
            "value": "beef",
          },
          "ref": null,
          "rendered": "beef",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "lamb",
            "value": "lamb",
          },
          "ref": null,
          "rendered": "lamb",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "seafood",
            "value": "seafood",
          },
          "ref": null,
          "rendered": "seafood",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "pork",
            "value": "pork",
          },
          "ref": null,
          "rendered": "pork",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "smoothies",
            "value": "smoothies",
          },
          "ref": null,
          "rendered": "smoothies",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "snacks",
            "value": "snacks",
          },
          "ref": null,
          "rendered": "snacks",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          id="dropdown"
          onChange={[Function]}
>
          <option
                    value="Categories"
          >
                    Categories
          </option>
          <option
                    value="breakfast"
          >
                    breakfast
          </option>
          <option
                    value="lunch"
          >
                    lunch
          </option>
          <option
                    value="dinner"
          >
                    dinner
          </option>
          <option
                    value="appetizer"
          >
                    appetizer
          </option>
          <option
                    value="chicken"
          >
                    chicken
          </option>
          <option
                    value="beef"
          >
                    beef
          </option>
          <option
                    value="lamb"
          >
                    lamb
          </option>
          <option
                    value="seafood"
          >
                    seafood
          </option>
          <option
                    value="pork"
          >
                    pork
          </option>
          <option
                    value="smoothies"
          >
                    smoothies
          </option>
          <option
                    value="snacks"
          >
                    snacks
          </option>
</select>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="Categories"
>
              Categories
</option>,
            <option
              value="breakfast"
>
              breakfast
</option>,
            <option
              value="lunch"
>
              lunch
</option>,
            <option
              value="dinner"
>
              dinner
</option>,
            <option
              value="appetizer"
>
              appetizer
</option>,
            <option
              value="chicken"
>
              chicken
</option>,
            <option
              value="beef"
>
              beef
</option>,
            <option
              value="lamb"
>
              lamb
</option>,
            <option
              value="seafood"
>
              seafood
</option>,
            <option
              value="pork"
>
              pork
</option>,
            <option
              value="smoothies"
>
              smoothies
</option>,
            <option
              value="snacks"
>
              snacks
</option>,
          ],
          "id": "dropdown",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Categories",
              "value": "Categories",
            },
            "ref": null,
            "rendered": "Categories",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "breakfast",
              "value": "breakfast",
            },
            "ref": null,
            "rendered": "breakfast",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "lunch",
              "value": "lunch",
            },
            "ref": null,
            "rendered": "lunch",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "dinner",
              "value": "dinner",
            },
            "ref": null,
            "rendered": "dinner",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "appetizer",
              "value": "appetizer",
            },
            "ref": null,
            "rendered": "appetizer",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "chicken",
              "value": "chicken",
            },
            "ref": null,
            "rendered": "chicken",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "beef",
              "value": "beef",
            },
            "ref": null,
            "rendered": "beef",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "lamb",
              "value": "lamb",
            },
            "ref": null,
            "rendered": "lamb",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "seafood",
              "value": "seafood",
            },
            "ref": null,
            "rendered": "seafood",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "pork",
              "value": "pork",
            },
            "ref": null,
            "rendered": "pork",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "smoothies",
              "value": "smoothies",
            },
            "ref": null,
            "rendered": "smoothies",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "snacks",
              "value": "snacks",
            },
            "ref": null,
            "rendered": "snacks",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
